tests/test_complete_edge_case_coverage.py

```
@@ -0,0 +1,158 @@
+                                                                assert config["api_url"] == "https://cli-url.com"
+                                                                assert config["model"] == "cli-model"
+
+        # Reset debug flag
+        git_commitai.DEBUG = original_debug
+
+    def test_get_git_diff_with_binary_file_dev_null(self):
+        """Test get_git_diff with binary file deleted or added."""
+        diff_output = "Binary files a/deleted.bin and /dev/null differ"
+
+        with patch("git_commitai.run_git", return_value=diff_output):
+            with patch("git_commitai.get_binary_file_info", return_value="Binary info"):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)